hooks/useGovernedMultiTypeAccounts.ts

```diff
@@ -0,0 +1,61 @@
+import { GovernedMultiTypeAccount } from '@utils/tokens'
+import { useCallback, useEffect, useState } from 'react'
+import useGovernanceAssets from './useGovernanceAssets'
+
+export default function useGovernedMultiTypeAccounts() {
+  const {
+    getMintWithGovernances,
+    governancesArray,
+    governedTokenAccounts,
+  } = useGovernanceAssets()
+
+  const [governedMultiTypeAccounts, setGovernedMultiTypeAccounts] = useState<
+    GovernedMultiTypeAccount[]
+  >([])
+
+  const getGovernedMultiTypeAccounts = useCallback(async (): Promise<
+    GovernedMultiTypeAccount[]
+  > => {
+    const mintWithGovernances = await getMintWithGovernances()
+
+    return governancesArray.map((gov) => {
+      const governedTokenAccount = governedTokenAccounts.find(
+        (x) => x.governance?.pubkey.toBase58() === gov.pubkey.toBase58()
+      )
+      if (governedTokenAccount) {
+        return governedTokenAccount as GovernedMultiTypeAccount
+      }
+
+      const mintGovernance = mintWithGovernances.find(
+        (x) => x.governance?.pubkey.toBase58() === gov.pubkey.toBase58()
+      )
+      if (mintGovernance) {
+        return mintGovernance as GovernedMultiTypeAccount
+      }
+
+      return {
+        governance: gov,
+      }
+    })
+  }, [JSON.stringify(governedTokenAccounts), JSON.stringify(governancesArray)])
```

Comment:
Stringifying because `governedTokenAccounts` & `governancesArray` aren't stable?

Reply:
exactly, it's a trick I've taken from existing code. If I don't, it just infinite refresh :( 

it's so bad lol

Reply:
Can you add a comment please:

```
FIXME: `governedTokenAccounts` & `governancesArray` should have stable references.
These should respect immutability principles & only change if their content changes.
Working around this by stringifying both objects & using the resulting string 
representation as hook dependency, so the hook only runs when either of these changes, 
but with a performance tax unfortunately . 
```

Reply:
Done